src/Components/Web.Extensions/src/InputFile/FileListEntryStream.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    internal abstract class FileListEntryStream : Stream
+    {
+        private long _position;
+
+        protected FileListEntry File { get; }
+
+        protected FileListEntryStream(FileListEntry file)
+        {
+            File = file;
+        }
+
+        public override bool CanRead => true;
+
+        public override bool CanSeek => false;
+
+        public override bool CanWrite => false;
+
+        public override long Length => File.Size;
+
+        public override long Position
+        {
+            get => _position;
+            set => throw new NotSupportedException();
+        }
+
+        public override void Flush()
+            => throw new NotSupportedException();
+
+        public override int Read(byte[] buffer, int offset, int count)
+            => throw new NotSupportedException("Synchronous reads are not supported.");
+
+        public override long Seek(long offset, SeekOrigin origin)
+            => throw new NotSupportedException();
+
+        public override void SetLength(long value)
+            => throw new NotSupportedException();
+
+        public override void Write(byte[] buffer, int offset, int count)
+            => throw new NotSupportedException();
+
+        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
```

Comment:
Should we also override the `Memory<byte>` overload?

Reply:
I've now added this, but it required some changes to `SharedBrowserFileStream` and `RemoteBrowserFileStream`. Please let me know if there's anything I should change to improve how the streams integrate with `Memory<byte>`.